modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,369 @@
+				 *                        Defaults to `current_filter()` when not set.
+				 * @param int $attachment_id the ID of the image attachment.
+				 *
+				 * @since 1.0.0
+				 */
+				$filtered_image = apply_filters( 'dominant_color_img_tag_add_adjust', $filtered_image, $context, $attachment_id );
+
+				if ( $filtered_image !== $match[0] ) {
+					$content = str_replace( $match[0], $filtered_image, $content );
+				}
+			}
+		}
+
+		return $content;
+	}
+
+	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
+	foreach ( $filters as $filter ) {
+		add_filter( $filter, 'dominant_color_filter_content_tags', 20 );
+	}
+
+	add_filter( 'dominant_color_img_tag_add_adjust', 'dominant_color_tag_add_adjust', 10, 3 );
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_add_inline_style() {
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominantcolor]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+
+/**
+ * Overloads wp_image_editors() to load the extended classes.
+ *
+ * @return string[]
+ */
+function dominant_color_set_image_editors() {
+
+	require_once 'class-dominant-color-image-editor-gd.php';
+	require_once 'class-dominant-color-image-editor-imagick.php';
+
+	return array( 'Dominant_Color_Image_Editor_GD', 'Dominant_Color_Image_Editor_Imagick' );
+}
+
+/**
+ * Get dominant color of image
+ *
+ * @param integer $id the image id.
+ * @param string  $default_color default color.
+ *
+ * @return string
+ * @since n.e.x.t
+ */
+function dominant_color_get( $id, $default_color = 'eee' ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file   = get_attached_file( $id );
+	$editor = wp_get_image_editor( $file );
+
+	if ( method_exists( $editor, 'get_dominant_color' ) ) {
+		$dominant_color = $editor->get_dominant_color( $default_color );
+	} else {
+		$dominant_color = $default_color;
+	}
+
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	return $dominant_color;
+}
+
+/**
+ * Works out if color has transparency
+ *
+ * @param integer $id the attachment id.
+ *
+ * @return bool
+ * @since n.e.x.t
+ */
+function dominant_color_get_has_transparency( $id ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file = get_attached_file( $id );
+
+	$editor = wp_get_image_editor( $file );
+	if ( is_wp_error( $editor ) || ! method_exists( $editor, 'get_has_transparency' ) ) {
+
+		return true; // safer to set to trans than not.
+	}
+	$has_transparency = $editor->get_has_transparency();
+
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
```

Comment:
I think the `remove_filter` function should be called right after calling the `wp_get_image_editor` function because otherwise if we get an error, we return `true` and don't remove filter:
```suggestion
	$editor = wp_get_image_editor( $file );
	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );

	if ( is_wp_error( $editor ) || ! method_exists( $editor, 'get_has_transparency' ) ) {
		return true; // safer to set to trans than not.
	}

	$has_transparency = $editor->get_has_transparency();
```